File: cpp/141A.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    string guest, host, pile;
    cin >> guest >> host >> pile;

    // Combine guest and host names
    string combined = guest + host;

    // Sort both strings
    sort(combined.begin(), combined.end());
    sort(pile.begin(), pile.end());

    // Compare
    if (combined == pile) cout << "YES\n";
    else cout << "NO\n";

    return 0;
}

File: java/141A.java
import java.util.*;

class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        String guest = sc.next();
        String host = sc.next();
        String pile = sc.next();

        // Combine guest + host
        String combined = guest + host;

        // Convert to char arrays for sorting
        char[] arr1 = combined.toCharArray();
        char[] arr2 = pile.toCharArray();

        Arrays.sort(arr1);
        Arrays.sort(arr2);

        // Compare sorted results
        if (Arrays.equals(arr1, arr2)) {
            System.out.println("YES");
        } else {
            System.out.println("NO");
        }
    }
}
